### PR TITLE
Initial workload split

### PR DESCRIPTION
**Pull request by lewing**
Work to move emscripten to it's own manifest.  Putting it up for discussion before it is finished.

Runtime PR https://github.com/dotnet/runtime/pull/54349

### CODE REVIEW

**Comment by lewing** [comment]
We don't need to do it now but we should move the emscripten setup/checks here as part of the base targets.

**Comment by lewing** [comment]
Now also generates the package name based off the properties not the file names.

cc @akoeplinger @radekdoulik

**Comment by lewing** [comment]
I'd like to start producing manifests asap for the reasons mentioned in dotnet/runtime#54349

**Review by steveisok — APPROVED** [review]

**Review by joeloff — APPROVED** [review]
Any changes needed to flow these packages to dotnet/runtime?

File: eng/nuget/Microsoft.NET.Workload.Emscripten.Manifest/Microsoft.NET.Workload.Emscripten.Manifest.pkgproj [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  <Import Project="$(NuGetPackageRoot)\microsoft.dotnet.build.tasks.packaging\$(MicrosoftDotNetBuildTasksPackagingVersion)\build\Microsoft.DotNet.Build.Tasks.Packaging.props" />
+  <Import Project="$(NuGetPackageRoot)\microsoft.dotnet.build.tasks.templating\$(MicrosoftDotNetBuildTasksTemplatingVersion)\build\Microsoft.DotNet.Build.Tasks.Templating.props" />
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Microsoft.NET.Runtime.Emscripten.Common.props))" />
+
+  <PropertyGroup>
+    <PackageDescription>Internal toolchain package not meant for direct consumption. Please do not reference directly.</PackageDescription>
+    <Description>Internal toolchain package not meant for direct consumption. Please do not reference directly.</Description>
+  </PropertyGroup>
+
```
Comment (radical) [inline eng/nuget/Microsoft.NET.Workload.Emscripten.Manifest/Microsoft.NET.Workload.Emscripten.Manifest.pkgproj:12]: extra new line

File: eng/nuget/Microsoft.NET.Workload.Emscripten.Manifest/WorkloadManifest.json.in [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  "packs": {
+    "Microsoft.NET.Runtime.Emscripten.Node" : {
+      "kind": "Sdk",
+      "version": "${PackageVersion}",
+      "alias-to": {
+        "win-x64": "Microsoft.NET.Runtime.${EmscriptenVersion}.Node.win-x64",
+        "linux-x64": "Microsoft.NET.Runtime.Emscripten.${EmscriptenVersion}.Node.linux-x64",
+        "osx-x64": "Microsoft.NET.Runtime.Emscripten.${EmscriptenVersion}.Node.osx-x64",
+        "osx-arm64": "Microsoft.NET.Runtime.Emscripten.${EmscriptenVersion}.Node.osx-x64"
```
Comment (radical) [inline eng/nuget/Microsoft.NET.Workload.Emscripten.Manifest/WorkloadManifest.json.in:23]: `osx-arm64` -> `osx-x64` - is this deliberate? I see this for the other two packs too

File: eng/nuget/Microsoft.NET.Workload.Emscripten.Manifest/WorkloadManifest.targets [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<Project>
```
Comment (radical) [inline eng/nuget/Microsoft.NET.Workload.Emscripten.Manifest/WorkloadManifest.targets:1]: 2 space indents like the other files

File: eng/nuget/Microsoft.NET.Runtime.Emscripten.Node/Microsoft.NET.Runtime.Emscripten.Node.pkgproj [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<Project>
+  <Import Project="$(RepoRoot)\Directory.Build.props" />
+  <Import Project="$(NuGetPackageRoot)\microsoft.dotnet.build.tasks.packaging\$(MicrosoftDotNetBuildTasksPackagingVersion)\build\Microsoft.DotNet.Build.Tasks.Packaging.props" />
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Microsoft.NET.Runtime.Emscripten.Common.props))" />
+  <Import Project="Microsoft.NET.Runtime.Emscripten.Node.props" />
+  
+  <Target Name="_PrepareForPack" BeforeTargets="GetPackageFiles" Returns="@(PackageFile)">
+    <!-- Override the id to include the sdk band as per the workload installer spec -->
+    <PropertyGroup>
+      <Id>Microsoft.NET.Runtime.Emscripten.$(EmscriptenVersion).Node.$(PackageRID)</Id>
```
Comment (radekdoulik) [inline eng/nuget/Microsoft.NET.Runtime.Emscripten.Node/Microsoft.NET.Runtime.Emscripten.Node.pkgproj:10]: Nice! I like that the files don't need to be renamed anymore.

**Review by radekdoulik — APPROVED** [review]

File: eng/nuget/Microsoft.NET.Workload.Emscripten.Manifest/WorkloadManifest.json.in (hunk 2 above)
Comment (lewing) [reply eng/nuget/Microsoft.NET.Workload.Emscripten.Manifest/WorkloadManifest.json.in:23]: This is to tell the sdk that is should use the osx-x64 packs when targeting osx-arm64 (they work using rosetta)

File: eng/Versions.props [hunk 5]
```diff
@@ -6,8 +6,10 @@
     <PreReleaseVersionIteration>7</PreReleaseVersionIteration>
     <!-- when bumping this you also need to rename the .pkgproj's in eng/nuget and eng/nuget/descriptions.json. TODO: make this better -->
```
Comment (akoeplinger) [inline eng/Versions.props:7]: this comment needs to be updated:

```suggestion
    <!-- when bumping this you also need to update eng/nuget/descriptions.json. TODO: make this better -->
```

Or can we get rid of descriptions.json too?

File: eng/nuget/Microsoft.NET.Runtime.Emscripten.Python/Microsoft.NET.Runtime.Emscripten.Python.pkgproj [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<Project>
+  <Import Project="$(RepoRoot)\Directory.Build.props" />
+  <Import Project="$(NuGetPackageRoot)\microsoft.dotnet.build.tasks.packaging\$(MicrosoftDotNetBuildTasksPackagingVersion)\build\Microsoft.DotNet.Build.Tasks.Packaging.props" />
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Microsoft.NET.Runtime.Emscripten.Common.props))" />
+  <Import Project="Microsoft.NET.Runtime.Emscripten.Python.props" />
+
+  <Target Name="_PrepareForPack" BeforeTargets="GetPackageFiles" Returns="@(PackageFile)">
+    <!-- Override the id to include the sdk band as per the workload installer spec -->
```
Comment (akoeplinger) [inline eng/nuget/Microsoft.NET.Runtime.Emscripten.Python/Microsoft.NET.Runtime.Emscripten.Python.pkgproj:8]: comment is wrong:

```suggestion
    <!-- Override the id to include the Emscripten version -->
```

File: eng/nuget/Microsoft.NET.Runtime.Emscripten.Node/Microsoft.NET.Runtime.Emscripten.Node.pkgproj [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<Project>
+  <Import Project="$(RepoRoot)\Directory.Build.props" />
+  <Import Project="$(NuGetPackageRoot)\microsoft.dotnet.build.tasks.packaging\$(MicrosoftDotNetBuildTasksPackagingVersion)\build\Microsoft.DotNet.Build.Tasks.Packaging.props" />
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Microsoft.NET.Runtime.Emscripten.Common.props))" />
+  <Import Project="Microsoft.NET.Runtime.Emscripten.Node.props" />
+
+  <Target Name="_PrepareForPack" BeforeTargets="GetPackageFiles" Returns="@(PackageFile)">
+    <!-- Override the id to include the sdk band as per the workload installer spec -->
```
Comment (akoeplinger) [inline eng/nuget/Microsoft.NET.Runtime.Emscripten.Node/Microsoft.NET.Runtime.Emscripten.Node.pkgproj:8]: comment is wrong:

```suggestion
    <!-- Override the id to include the Emscripten version -->
```

File: eng/nuget/Microsoft.NET.Runtime.Emscripten.Sdk/Microsoft.NET.Runtime.Emscripten.Sdk.pkgproj [hunk 8]
```diff
@@ -0,0 +1,21 @@
+<Project>
+  <Import Project="$(RepoRoot)\Directory.Build.props" />
+  <Import Project="$(NuGetPackageRoot)\microsoft.dotnet.build.tasks.packaging\$(MicrosoftDotNetBuildTasksPackagingVersion)\build\Microsoft.DotNet.Build.Tasks.Packaging.props" />
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Microsoft.NET.Runtime.Emscripten.Common.props))" />
+  <Import Project="Microsoft.NET.Runtime.Emscripten.Sdk.props" />
+
+  <Target Name="_PrepareForPack" BeforeTargets="GetPackageFiles" Returns="@(PackageFile)">
+    <!-- Override the id to include the sdk band as per the workload installer spec -->
```
Comment (akoeplinger) [inline eng/nuget/Microsoft.NET.Runtime.Emscripten.Sdk/Microsoft.NET.Runtime.Emscripten.Sdk.pkgproj:8]: comment is wrong:

```suggestion
    <!-- Override the id to include the Emscripten version -->
```

**Review by akoeplinger — COMMENTED** [review]
Left a few comments, looks good otherwise :)

File: eng/Versions.props (hunk 5 above)
Comment (lewing) [reply eng/Versions.props:7]: I think we can get rid of it

**Pull request merged**